consensus/hotstuff/signature/weighted_signature_aggregator.go

```diff
@@ -1,39 +1,141 @@
 package signature
 
 import (
-	"github.com/onflow/flow-go/consensus/hotstuff"
+	"fmt"
+	"sync"
+
 	"github.com/onflow/flow-go/crypto"
 	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/signature"
 )
 
-// WeightedSignatureAggregator aggregates the signatures
+// WeightedSignatureAggregator implements consensus/hotstuff.WeightedSignatureAggregator
 type WeightedSignatureAggregator struct {
+	*signature.SignatureAggregatorSameMessage                              // low level crypto aggregator, agnostic of weights and flow IDs
```

Comment:
How about putting it as member? I find it quite confusing since methods are named exactly the same.

Reply:
Same [comment](https://github.com/onflow/flow-go/pull/1292#discussion_r708493745)